src/rect.rs

```diff
@@ -421,6 +421,15 @@ impl Rect {
         RoundedRect::from_rect(self, radius)
     }
 
+    /// The aspect ratio of the `Rect`.
+    ///
+    /// This is defined as the height divided by the width. It measures the "squareness" of the
+    /// rectangle (a value of 1 is square).
+    #[inline]
+    pub fn aspect_ratio(&self) -> f64 {
+        (self.y1 - self.y0) / (self.x1 - self.x0)
```

Comment:
an observation: I find with methods like this if I can define them in one place and then call out to that I prefer that approach, and so I'd have probably just done `self.size().aspect_ratio()`.